backends/dpdk/dpdkHelpers.cpp

```diff
@@ -346,6 +437,166 @@ bool ConvertStatementToDpdk::preorder(const IR::AssignmentStatement *a) {
+                        auto rt = new IR::Member(new IR::PathExpression(IR::ID("h")), name);
+                        auto lt = new IR::Member(new IR::PathExpression(IR::ID("m")), fldName);
+                        BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                        metadataStruct->fields.push_back(new IR::StructField(fldName, it->type));
+                        add_instr(new IR::DpdkMovStatement(lt, rt));
+                        components.push_back(lt);
+                    }
+                } else {
+                    IR::ID name(refmap->newName(getHdrMdStrName(params) + "_" +
+                                                params->member.name));
+                    BUG_CHECK(metadataStruct, "Metadata structure missing unexpectedly!");
+                    metadataStruct->fields.push_back(new IR::StructField(name, params->type));
+                    auto lt = new IR::Member(new IR::PathExpression(IR::ID("m")), name);
+                    add_instr(new IR::DpdkMovStatement(lt, params));
+                    components.push_back(lt);
+                }
+            }
+        }
+    }
+}
+
+/* This function returns the header/metadata structure name */
+cstring ConvertStatementToDpdk::getHdrMdStrName(const IR::Member* mem) {
+    cstring sName = "";
+    if ((mem != nullptr) && (mem->expr != nullptr) &&
+       (mem->expr->type != nullptr)) {
+        if (auto st = mem->expr->type->to<IR::Type_Header>()) {
+            sName = st->name.name;
+        } else if (auto st = mem->expr->type->to<IR::Type_Struct>()) {
+             sName = st->name.name;
+        }
+    }
+    return sName;
+}
+
+/* This function processes Hash parameters and checks if all parameters
+   belong to same header/metadata structure.
+*/
+bool ConvertStatementToDpdk::checkIfBelongToSameHdrMdStructure(const IR::Argument* field) {
+    if (auto s = field->expression->to<IR::StructExpression>()) {
+        if (s->components.size() == 1)
+            return true;
+
+        cstring hdrStrName = "";
+        for (auto field1 : s->components) {
+            cstring sName = "";
+            if (auto params = field1->expression->to<IR::Member>()) {
+                auto type = typemap->getType(params, true);
+                if (type->is<IR::Type_Header>()) {
+                    auto typeheader = type->to<IR::Type_Header>();
+                    sName = typeheader->name.name;
+                } else if ((params != nullptr) && (params->expr != nullptr) &&
+                           (params->expr->type != nullptr)) {
+                    sName = getHdrMdStrName(params);
+                }
+            }
+            if (hdrStrName == "")
+                hdrStrName = sName;
+            else if (hdrStrName != sName)
+                return false;
+        }
+    }
+    return true;
+}
+
+/* This function processes hash parameters and checks if all parameters
+   are contiguous field or not in a header/metadata structure.
```

Comment:
I would refine this comment to explain when this function returns true.

Reply:
Updated